src/App.vue

```diff
@@ -1,14 +1,104 @@
 <template>
-  <Base/>
+  <v-app light>
+    <Navbar :menu-items="menuItems" />
+    <MobileNavbar :menu-items="menuItems" />
+    <vue-page-transition style="height:100%" >
+    <component :is="layout">
+      <vue-page-transition style="height:100%" >
+        <router-view/>
+      </vue-page-transition>
+    </component>
+    </vue-page-transition>
+  </v-app>
 </template>
 
 <script>
-import Base from './views/Base'
+import Default from '@/components/layouts/Default.vue'
+import Large from '@/components/layouts/Large.vue'
+import None from '@/components/layouts/None.vue'
+import Navbar from '@/components/Navbar.vue'
+import MobileNavbar from '@/components/MobileNavbar.vue'
+
+import { between, required, email } from 'vee-validate/dist/rules'
+import { extend } from 'vee-validate'
+
+extend('passwordMatch', {
+  validate: (value, { other }) => value === other,
+  message: 'Hasła nie są identyczne.',
+  params: [{ name: 'other', isTarget: true }]
+})
+extend('between', {
+  validate: (value, { min, max }) => (value.length > min && value.length < max),
+  message: '{_field_} musi składać się z więcej niż {min} znaków.',
+  params: between.params
+})
+
+extend('require', {
+  validate: required.validate,
+  computesRequired: required.computesRequired,
+  message: 'Pole {_field_} jest wymagane.'
+})
+
+extend('email', {
+  validate: email.validate,
+  message: 'Podany adres email jest niepoprawny.'
+})
 
 export default {
   name: 'App',
   components: {
-    Base
+    Default,
+    Large,
+    None,
+    Navbar,
+    MobileNavbar
+  },
+  data () {
+    return {
+      menuItems: [
+        {
+          title: 'Rejestracja',
+          path: { name: 'RegistrationForm' },
+          icon: 'Face',
+          requireAuth: false
+        },
+        {
+          title: 'Logowanie',
+          path: { name: 'LoginForm' },
+          icon: 'LockOpen',
+          requireAuth: false
+        },
+        {
+          title: 'Wyloguj się',
+          path: { name: 'Home' },
+          icon: 'Logout',
+          requireAuth: true
+        },
+        {
+          title: 'Katalog Quizów',
+          path: { name: 'QuizList', params: { pageNumber: 1 } },
+          icon: 'BookMultiple',
+          requireAuth: false
+        }
+      ]
+    }
+  },
```

Comment:
To nie powinno być w navbarze?

Reply:
Te dane wykorzystywane są przez dwa komponenty Navbar i MobileNavbar. Aby uniknąć duplikacji kodu i możliwości wprowadzenia rozbieżności w nawigacji postanowiłem przekazywać te dane w propie z jednego miejsca. Myślałem o przeniesieniu tych danych do vuexa, ale wykorzystywane byłyby tylko gettery więc nie wiem czy to dobra praktyka. Ewentualnie można utworzyć base layout ze slotem na implementację layoutu i wrzucić tam tworzenie navbaru i mobile navbaru razem z tymi danymi. Co o tym sądzisz? Masz może jakieś inne pomysły?

Reply:
Gettery są po to żeby z nich korzystać. Są trzy dobre rozwiązania moim zdaniem tej sytuacji.
1) Robisz component Menu.vue i rozszerzasz NavbarDesktop i NavbarMobile.
2) Przechowujesz te dane w storze i sobie je pobierasz po prostu getterem
3) Skorzystaj z responsywności, przełączania klas etc.

Reply:
Skorzystałem z opcji 2